Tests/SwiftFusionTests/Inference/ProbablisticTrackingFactorTests.swift

```diff
@@ -0,0 +1,65 @@
+// Copyright 2020 The SwiftFusion Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import TensorFlow
+import SwiftFusion
+import XCTest
+
+final class ProbablisticTrackingFactorTests: XCTestCase {
+  func testCreation() {
+    var v = VariableAssignments()
+    let poseId = v.store(Pose2())
+
+    let _ = ProbablisticTrackingFactor(poseId,
+      measurement: Tensor<Double>([[0]]),
+      encoder: PPCA(latentSize: 10),
+      patchSize: (10, 10),
+      appearanceModelSize: (10, 1),
+      foregroundModel: GaussianNB(dims: [10]),
+      backgroundModel: GaussianNB(dims: [10])
+    )
+  }
+
+  func testErrorSanity() {
+    var v = VariableAssignments()
+    let pose = Pose2(3, 3, 0)
+    let poseId = v.store(pose)
+
+    var image: Tensor<Double> = .init(zeros: [6, 6, 3])
+
+    image[3, 3, 1] = Tensor(1.0)
+
+    var encoder = PPCA(latentSize: 1)
```

Comment:
I'd assign a variable d=10 here. The fact that you can give an encoder below with d=1 and generative models with d-10 means that you might want a precondition in constructor. But hey, after this small comment just merge this PR and then make the unit test work :-)